node/src/actors/chain_manager/transaction_factory.rs

```diff
@@ -53,7 +53,7 @@ pub fn take_enough_utxos(
         }
 
         if let Some(block_number_limit) = block_number_limit {
-            if all_utxos.included_in_block_number(op).unwrap() >= block_number_limit {
+            if all_utxos.included_in_block_number(op).unwrap() > block_number_limit {
```

Comment:
BTW, are we totally sure that `block_number_limit` is always greater than `0` ??? :thinking:

Reply:
block_number_limit is not always > 0, that it is the result of (current_epoch - collateral_age). If collateral_age is bigger than current_epoch, block_number_limit will be 0. Anyways, the important number is the result of 'included_in_block_number' that in the case of genesis is 0, and 0 never will be bigger than the minimum block_number_limit (0)

Reply:
Cool, what about this?
```suggestion
            // Ignore all outputs created after `block_number_limit`.
            // Outputs from the genesis block will never be ignored because `block_number_limit` can't go lower than `0`.
            if all_utxos.included_in_block_number(op).unwrap() > block_number_limit {
```

Reply:
// Ignore outputs created after `block_number_limit`.
            // Outputs from the genesis block will never be ignored because `block_number_limit` is never smaller than `0`.